lib/screens/register.dart

```diff
@@ -34,61 +42,106 @@ class RegisterScreenState extends State<RegisterScreen> {
                 ),
-                // errorText: _errorMessage.isNotEmpty ? _errorMessage : null,
+                counterText: '',
               ),
-              textAlign: TextAlign.center,
             ),
+
             const SizedBox(height: 20),
+
             TextField(
-              controller: _usernameController,
-              decoration: const InputDecoration(
-                label: Center(
-                  child: Text("USERNAME"),
+              controller: _emailController,
+              onChanged: (text) {
+                _emailController.text = text.toLowerCase();
+                setState(() {
+                  validEmail = _validEmail(_emailController.text);
+                });
+              },
+              textAlign: TextAlign.center,
+              decoration: InputDecoration(
+                label: const Center(
+                  child: Text('EMAIL'),
                 ),
-                // errorText: _errorMessage.isNotEmpty ? _errorMessage : null,
+                errorMaxLines: 2,
+                errorText: validEmail ? null : 'Please enter a valid email address',
               ),
-              textAlign: TextAlign.center,
             ),
+
             const SizedBox(height: 20),
+
             TextField(
               controller: _passwordController,
+              onChanged: (text) {
+                setState(() {
+                  validPassword = _validPassword(_passwordController.text);
+                });
+              },
               textAlign: TextAlign.center,
-              decoration: const InputDecoration(
-                label: Center(
-                  child: Text("PASSWORD"),
+              decoration: InputDecoration(
+                label: const Center(
+                  child: Text('PASSWORD'),
                 ),
+                errorMaxLines: 4,
+                errorText: validPassword ? null : 'Passwords need a number, special character, lowercase and uppercase letter',
               ),
               obscureText: true,
             ),
+
             const SizedBox(height: 20),
+
             TextField(
-              controller: _passwordController,
+              controller: _repeatPasswordController,
               textAlign: TextAlign.center,
               decoration: const InputDecoration(
                 label: Center(
-                  child: Text("REPEAT PASSWORD"),
+                  child: Text('REPEAT PASSWORD'),
                 ),
               ),
               obscureText: true,
             ),
+            const SizedBox(height: 100),
           ],
         ),
       ),
+
       floatingActionButton: Column(
         mainAxisSize: MainAxisSize.min,
         mainAxisAlignment: MainAxisAlignment.end,
         children: [
           const SizedBox(height: 20),
           ElevatedButton(
-            onPressed: onTest,
-            child: const Text('VERIFY'),
+            onPressed: validEmail || validPassword ? onRegister : null,
```

Comment:
this should be `validEmail && validPassword`. as is, i'm able to send a broken email to the backend if I have a valid password, and viceversa